### PR TITLE
More lazy fixes and preprocessing functions

### PR DESCRIPTION
**Pull request by bouweandela**
## Description

Various small improvements in lazy data computations by using coordinate methods `has_bounds()`, `core_points()`, `core_bounds()` where possible.

Related to #674 and #42

***

## [Before you get started](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#getting-started)

- [x] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#code-quality)
- [x] [🛠][1] [Unit tests](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#tests) have been added
- [x] [🛠][1] Changes are [backward compatible](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility)
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#list-of-authors) is up to date
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

File: esmvalcore/cmor/_fixes/cmip5/hadgem2_cc.py [hunk 1]
```diff
@@ -1,34 +1,10 @@
...
-        for cube in cubes:
-            lats = cube.coords('latitude')
-            if lats:
-                lat = cube.coord('latitude')
-                lat.points = np.clip(lat.points, -90., 90.)
-                lat.bounds = np.clip(lat.bounds, -90., 90.)
-
-        return cubes
+class AllVars(BaseAllVars):
+    """Fix errors common to all vars."""
```
Comment (schlunma) [inline esmvalcore/cmor/_fixes/cmip5/hadgem2_cc.py:7]:
```suggestion
AllVars = BaseAllVars
```

File: esmvalcore/preprocessor/_units.py [hunk 2]
```diff
@@ -120,7 +121,10 @@ def convert_units(cube, units):
     return cube
 
 
-def accumulate_coordinate(cube, coordinate):
+def accumulate_coordinate(
+    cube: iris.cube.Cube,
+    coordinate: str,
```
Comment (schlunma) [inline esmvalcore/preprocessor/_units.py:126]: `coordinate` could also be in `iris.coords.Coord`. Please also remove the type hints from the docstring below, they will get added automatically

**Review by schlunma — APPROVED** [review]
Thanks Bouwe, LGTM! Just some minor comments from my side.

File: esmvalcore/preprocessor/_units.py (hunk 2 above)
Comment (bouweandela) [reply esmvalcore/preprocessor/_units.py]: Done in [b3d0f76](https://github.com/ESMValGroup/ESMValCore/pull/2325/commits/b3d0f7637922b54464b54bbfefd34830fb4661ea)

File: esmvalcore/cmor/_fixes/cmip5/hadgem2_cc.py (hunk 1 above)
Comment (bouweandela) [reply esmvalcore/cmor/_fixes/cmip5/hadgem2_cc.py:7]: Not too sure about this one, why is it better to write it like this?

Comment (bouweandela) [reply esmvalcore/cmor/_fixes/cmip5/hadgem2_cc.py:7]: Anyway, is be happy to change it, just curious

Comment (schlunma) [reply esmvalcore/cmor/_fixes/cmip5/hadgem2_cc.py:7]: It's shorter 😁 Both versions are fine for me, your call

**Comment by valeriupredoi** [comment]
brill work, fellas :beers:

**Pull request merged**